### PR TITLE
Adding getServerRuntime to RuntimePrefPage

### LINKED ISSUE #1711: RuntimePreferencePage is missing getRuntime

### PR DESCRIPTION
**Pull request by odockal**
- fixes #1711

Signed-off-by: Ondrej Dockal <odockal@redhat.com>

### CODE REVIEW

File: plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/wst/server/ui/RuntimePreferencePage.java [hunk 1]
```diff
@@ -71,6 +72,47 @@ public RuntimePreferencePage(ReferencedComposite referencedComposite) {
...
+	 * @return runtime object if equals to passed parameter, null if there is none or
+	 *  throws an exception if there are more than one runtime found
+	 */
+	public Runtime getServerRuntime(String name) {
+		List<Runtime> resultRuntimes = getServerRuntimes()
+				.stream()
+				.filter((runtime) -> runtime.getName().equals(name))
+				.collect(Collectors.toList());
+		if (resultRuntimes.size() > 1) {
+			throw new IllegalStateException("There are more than one runtime fulfilling conditions. Result is ambiguous.");
```
Comment (rawagner) [inline plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/wst/server/ui/RuntimePreferencePage.java]: throw EclipseLayerEx

File: tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/wst/server/ui/TestServerRuntime2.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc and others.
```
Comment (rawagner) [inline tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/wst/server/ui/TestServerRuntime2.java]: 2018

File: tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/wst/server/ui/RuntimePreferencePageTest.java [hunk 3]
```diff
@@ -111,4 +114,39 @@ public void editRuntime(){
...
+		Runtime runtime1 = preferencePage.getServerRuntime(SERVER_NAME, SERVER_NAME);
+		assertThat(runtime1, is(new Runtime(SERVER_NAME, SERVER_NAME)));
+		
+		Runtime runtime2 = preferencePage.getServerRuntime(new Runtime(SERVER_NAME_2, SERVER_NAME_2));
+		assertThat(runtime2, is(new Runtime(SERVER_NAME_2, SERVER_NAME_2)));
+		
+		Runtime runtime3 = null;
+		try {
+			runtime3 = preferencePage.getServerRuntime(SERVER_NAME_2);
+		} catch (IllegalStateException exc) {
```
Comment (rawagner) [inline tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/wst/server/ui/RuntimePreferencePageTest.java]: I guess EclipseLayerEx should be in the catch block

Comment (odockal) [reply tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/wst/server/ui/RuntimePreferencePageTest.java]: Oh dear, you are right... I have not had a coffee yet... Thanks

**Comment by odockal** [comment]
@rawagner I have changed property name and another test has failed, will repair now and re-run tests.

**Comment by odockal** [comment]
@rawagner Some other tests failed, so this seems alright.

**Pull request merged**